api/services/tournamentService.js

```diff
@@ -1129,23 +1131,23 @@ const updateMatchResult = async (uid, matchID, score1, score2) => {
     }
 
     // Fetch the match using matchID
-    const match = await searchMatches(matchID);
+    const [match] = await TournamentModel.searchMatches(matchID);
 
     // Ensure the match exists
     if (!match) throw new Error("Could not find match");
 
     // Get participant information for Team1 to identify round and bracket side
-    const participants = await searchTournamentParticipants(
+    const [participants] = await TournamentModel.searchTournamentParticipants(
         match.TournamentID,
         match.Team1ID
     );
 
     // Ensure participant information is available
-    if (!participants || participants.length === 0)
+    if (!participants)
         throw new Error("Could not retrieve participant info for Team1.");
```

Comment:
Again, attempting to use httperrors as much as possible to make bugs easier to diagnose (not sure if 404 is the best fit here tho)
```suggestion
        throw createHttpError(404, "Could not retrieve participant info for Team1.");
```